### ISSUE TITLE
Addon-docs: 'show code' not displaying correct component name when using forwardRef with args

### ISSUE THREAD

**Issue by Jamess-Lucass** [storybookjs/storybook#18972]
**Describe the bug**
When using args with a component that uses `forwardRef`, you will not be shown the correct component name when clicking 'Show code' in the docs section of storybook

**To Reproduce**
Repo: https://github.com/Jamess-Lucass/storybook-issue-reproduction

1. Create a component using `forwardRef`
2. Use that component with the args feature within storybook

**System**
```bash
Environment Info:

  System:
    OS: Linux 5.10 Ubuntu 20.04.4 LTS (Focal Fossa)
    CPU: (32) x64 AMD Ryzen 9 5950X 16-Core Processor
  Binaries:
    Node: 16.16.0 - ~/.nvm/versions/node/v16.16.0/bin/node
    npm: 8.11.0 - ~/.nvm/versions/node/v16.16.0/bin/npm
  npmPackages:
    @storybook/addon-actions: 6.5.10 => 6.5.10
    @storybook/addon-essentials: 6.5.10 => 6.5.10
    @storybook/addon-interactions: 6.5.10 => 6.5.10
    @storybook/addon-links: 6.5.10 => 6.5.10
    @storybook/builder-vite: 0.2.2 => 0.2.2
    @storybook/react: 6.5.10 => 6.5.10
    @storybook/testing-library: 0.0.13 => 0.0.13
```

**Additional context**
I am experiencing this issue when the component uses `forwardRef` and your story is using args. The below code will replicate this error.

currently on version `6.5.10`

`button.tsx` - normal component
```typescript
import { forwardRef } from "react";

export type ButtonProps = {
  text: string;
};

const Button = forwardRef<HTMLButtonElement, ButtonProps>(
  ({ text }: ButtonProps, ref) => <button ref={ref}>{text}</button>
);

// const Button = ({ text }: ButtonProps) => <button>{text}</button>;

export default Button;
```

`button.stories.tsx` - story file
```typescript
import { ComponentMeta, ComponentStory } from "@storybook/react";
import Button from "./button";

export default {
  title: "Button",
  component: Button,
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = ({ text }) => (
  <Button text={text} />
);

// const Template: ComponentStory<typeof Button> = () => (
//   <Button text="button text" />
// );

export const Default = Template.bind({});
Default.args = {
  text: "Best button!",
};
```

When viewing the docs in storybook you will see the component loaded as `_c`:

![storybook-forwardref-template-args-issue](https://user-images.githubusercontent.com/23193271/185718860-d067ba16-551d-4b87-ac3c-0d74cf9b0800.png)

If you remove `args` in the `button.stories.tsx` file it will work:
```typescript
const Template: ComponentStory<typeof Button> = () => <Button text="test" />;
```
This is not feasible for me as I want to use the args feature.

Likewise, if you were to not use `forwardRef` in the `button.tsx` component it will work with args:
```typescript
const Button = ({ text }: ButtonProps) => <button>{text}</button>;
```
This is also not feasible for me as I want to use `forwardRef` in the component and others may not have direct access to the component if they're using a library like MUI or Chakra

**Comment by zhyd1997** [comment]
can not reproduce with `builder-webpack5`:

```shell
# setup react app
npx create-react-app example-webpack5 --template typescript
# switch to app
cd example-webpack5
# setup storybook
npx storybook init
# run storybook
npm run storybook
```

<img width="1020" alt="Screen Shot 2022-08-20 at 12 23 52" src="https://user-images.githubusercontent.com/31362988/185728823-62962f7b-93cd-47c2-927c-f112dff35e25.png">

<img width="439" alt="Screen Shot 2022-08-20 at 12 18 38" src="https://user-images.githubusercontent.com/31362988/185728710-cc3187c3-48f3-498d-9f4b-8c32ea13c3a0.png">

**Comment by zhyd1997** [comment]
**Update**:

@Jamess-Lucass 

`button` component must be capitalized - `Button.tsx` works for me.

**Comment by zhyd1997** [comment]
`button` will also break with `builder-webpack5`, not sure if it's a bug 👀

**Comment by Jamess-Lucass** [comment]
Hi @zhyd1997

I followed your steps above and capitalizing the component name has seemed to fix it. 

I tried capitalizing the component in my `@storybook/builder-vite` storybook project and it did not work, so doesn't seem to be an issue with storybook itself.

I will try a few bits to get it working in my current project and will make another comment when a solution has been found in case anyone runs into this

**Comment by zhyd1997** [comment]
@Jamess-Lucass 

Yes, you're right, the difference between the 2 name conventions is if triggered `isForwardRef` condition, I am not sure if it's a bug 🐛.

**Comment by zhyd1997** [comment]
I guess the `_c` is generated here:
https://github.com/storybookjs/storybook/blob/f2324595c8953682dec3be6a3201aca577ad9467/lib/core-common/src/utils/to-importFn.ts#L34-L39

which is called by `importFn`:
https://github.com/storybookjs/storybook/blob/f2324595c8953682dec3be6a3201aca577ad9467/lib/store/src/StoryStore.ts#L125

**Comment by zhyd1997** [comment]
Maybe this is an issue related to Webpack (both for `builder-webpack4` and `builder-webpack5`)? 👀 

<img width="474" alt="Screen Shot 2022-08-21 at 22 51 07" src="https://user-images.githubusercontent.com/31362988/185797070-0490f548-d97b-48fd-8d35-9f8407798333.png">

can be reproduced **without** the storybook setup.

**Comment by zhyd1997** [comment]
Note: here the `button.tsx` used **default export**, if changed it to **named export**, it can also solve this issue.

**Comment by warman-suganda** [comment]
Hi, I have the same issue with `forwardRef`. I'm not sure if this is the best way, but it works for me. I wrap the original component first before using that component on the playground. e.g:

**button.stories.tsx - story file**
```tsx
import { ComponentMeta } from '@storybook/react';
import ButtonComponent, { ButtonProps } from "./button";

/* Wrapper */
function Button(props: ButtonProps) {
  return <ButtonComponent {...props} />;
}

export default {
  title: 'Button',
  component: Button,
} as ComponentMeta<typeof Button>;

function Template({ text }: ButtonProps) {
  return <Button text={text} />;
}

export const Default = Template.bind({});

Default.args = {
  text: 'Best button!',
};
```
<img width="1310" alt="Screenshot 2023-01-02 at 23 28 18" src="https://user-images.githubusercontent.com/110805068/210258394-3d8f8d84-31a5-47c2-842c-a291a2686da7.png">

Please let me know if there is a better solution. Thank you.

**Comment by XQY279** [comment]
I have the same problem.

**Comment by william-will-angi** [comment]
@warman-suganda's suggestion worked for me, but only in dev mode 😢

**Comment by BAHAA-THE-KING** [comment]
I am having the same problem with MUI Icon, it displays as Icon2 instead of Icon (and many other components).
Any Updates ??

**Comment by binomialstew** [comment]
Instead of creating a new component wrapper, you can set the `displayName` of the component:
`Component.displayName = 'Component';`. Unfortunately, I haven't found a config option to prevent this name replacement.

**Comment by recallwei** [comment]
If I use
`export const MyComponent = ()=> <></>`
I cannot set displayName, maybe I only use `function` to declare a component, it can do that.

**Comment by just-be-weird** [comment]
Any updates on this? It's still happening with latest version of storybook. Setting displayName in every file for every component does not seem right. 

```jsx
    <React.ForwardRef
      sx={{
        alignItems: 'center',
        bgcolor: '#F1F1F1',
        display: 'flex',
        height: '100%',
        justifyContent: 'center'
      }}
    >
      <React.ForwardRef
        sx={{
          color: '#A8A29E'
        }}
        variant="body1"
      >
        Page goes here
      </React.ForwardRef>
```

After doing this 

Box.displayName = "Box"
Container.displayName = "Container"
Typography.displayName = "Typography"

issue seem to be resolved on local but not on prod
